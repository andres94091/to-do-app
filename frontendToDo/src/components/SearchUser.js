import React, { useEffect, useState } from 'react'
import axios from 'axios'

export const SearchUser = ({setUserHook}) => {

  const [arrayUsersTasks, setArrayUsersTasks] = useState([])

  useEffect(async () => {
    const response = await axios.get(`http://localhost:5000/api/users`)
    const { data: usersData } = response.data
    setArrayUsersTasks(usersData)
  }, [])

  const renderUserOptions = () => {
    return arrayUsersTasks.map((user, index) => {
      return (
        <option value={user.id}>{user.name}</option>
      )
    })
  }

  const handleUserSelect = (event) => {
    setUserHook(event.target.value);
  }

  return (
    <>
      <select className= "custom-select custom-select-lg" list = "usersList" onChange={handleUserSelect}>
        {renderUserOptions()}
      </select>
    </>
  )
}
