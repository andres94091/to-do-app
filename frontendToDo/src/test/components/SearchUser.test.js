import '@testing-library/jest-dom';

import React from 'react';
import { shallow } from 'enzyme';
import SearchUser from '../../components/SearchUser';

const setUserHook = jest.fn();

describe('test in <SearchUser/>', () => {

  const wrapper = shallow(<SearchUser
    setUserHook = {setUserHook}
  />)

  it('should render the component correctly', () => {

    expect(wrapper).toMatchSnapshot();

  })
  
  
})
