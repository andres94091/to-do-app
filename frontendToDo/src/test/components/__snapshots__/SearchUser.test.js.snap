// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <SearchUser/> should render the component correctly 1`] = `
<Fragment>
  <select
    className="custom-select custom-select-lg"
    list="usersList"
    onChange={[Function]}
  />
</Fragment>
`;
